criu/mount.c

```diff
@@ -487,19 +487,16 @@ static int try_resolve_ext_mount(struct mount_info *info)
 
 		val = external_lookup_by_key(devstr);
 		if (!IS_ERR_OR_NULL(val)) {
-			char *source;
 			int len;
 
 			len = strlen(val) + sizeof("dev[]");
-			source = xmalloc(len);
-			if (source == NULL)
+			info->source = xrealloc(info->source, len);
+			if (info->source == NULL)
```

Comment:
realloc() doesn't touch original memory if re-allocation failed
That means that info->source original pointer will leak now.
It's not that we're precise about alloc/free in case of error in criu..
But don't see the reason to add additional leak for a fallback.
Otherwise, the patch makes sense to me.

Reply:
> realloc() doesn't touch original memory if re-allocation failed
That means that info->source original pointer will leak now.

Thanks for catching that, I will update the commit to fix it.

> It's not that we're precise about alloc/free in case of error in criu..
But don't see the reason to add additional leak for a fallback.

If we are to be precise about alloc/free in case of error then
we could use tools like Valgrind to check for memory leaks.